tendrl/alerting/handlers/__init__.py

```diff
@@ -0,0 +1,224 @@
+                    {
+                        "message": 'Failed to handle the alert of resource %s'
+                        ' for node %s. Error %s' % (
+                            alert_obj.resource,
+                            alert_obj.node_id,
+                            str(ex)
+                        )
+                    }
+                )
+            )
+
+
+class AlertHandlerManager(multiprocessing.Process):
+    def load_handlers(self):
+        try:
+            path = os.path.dirname(os.path.abspath(__file__))
+            pkg = 'tendrl.alerting.handlers'
+            for file_name in os.listdir(path):
+                component_path = os.path.join(path, file_name)
+                if os.path.isdir(component_path):
+                    for py in [f[:-3] for f in os.listdir(component_path)
+                               if f.endswith('.py') and f != '__init__.py']:
+                        handler_name = '.'.join([pkg, file_name, py])
+                        mod = importlib.import_module(handler_name)
+                        clsmembers = inspect.getmembers(mod, inspect.isclass)
+                        for name, cls in clsmembers:
+                            if cls.handles:
+                                exec(
+                                    "from %s import %s" % (
+                                        handler_name,
+                                        name
+                                    )
+                                )
+                                self.alert_handlers.append(cls.handles)
+        except (SyntaxError, ValueError, ImportError) as ex:
+            Event(
+                Message(
+                    "error",
+                    "alerting",
+                    {
+                        "message": 'Failed to load the alert handlers.'
+                        'Error %s' % (
+                            str(ex)
+                        )
+                    }
+                )
+            )
+            raise ex
+
+    def __init__(self):
+        super(AlertHandlerManager, self).__init__()
+        try:
+            self.alert_handlers = []
+            self.complete = multiprocessing.Event()
+            self.load_handlers()
+            self.init_alerttypes()
+        except (SyntaxError, ValueError, ImportError) as ex:
+            Event(
+                Message(
+                    "error",
+                    "alerting",
+                    {
+                        "message": 'Error %s' % (
+                            str(ex)
+                        )
+                    }
+                )
+            )
+            raise ex
+
+    def init_alerttypes(self):
+        for handler in AlertHandler.handlers:
+            tendrl_ns.alert_types.append(handler.representive_name)
+
+    def run(self):
+        try:
+            while not self.complete.is_set():
+                new_msg_id = tendrl_ns.alert_queue.get()
+                time.sleep(15)
```

Comment:
No this should not be configurable. This is a latency added only for some init in libraries to complete because of which we used to see some issues.
Like the urllib3 request was made before its pool was created and some such